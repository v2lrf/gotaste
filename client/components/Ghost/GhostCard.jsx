import React from 'react'
import PropTypes from 'prop-types'

function GhostCard({ height, width }) {
  return (
    <div
      className="rounded shimmer"
      style={{
        height: `${height}px`,
        width: `${width}%`
      }}
      aria-disabled
      aria-label="Henter data"
    />
  )
}

GhostCard.propTypes = {
  height: PropTypes.number.isRequired,
  width: PropTypes.number
}

GhostCard.defaultProps = {
  width: 100
}

export default GhostCard
