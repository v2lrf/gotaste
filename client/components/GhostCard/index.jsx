import React from 'react'
import PropTypes from 'prop-types'
import ContentLoader from 'react-content-loader'

function GhostCard({ height, width }) {
  return (
    <ContentLoader
      height={height}
      width={width}
      speed={2}
      primaryColor="#f3f3f3"
      secondaryColor="#ecebeb"
      className="mb-4"
    >
      <rect rx="4" ry="4" width={width} height={height} />
    </ContentLoader>
  )
}

GhostCard.propTypes = {
  height: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired
}

export default GhostCard
