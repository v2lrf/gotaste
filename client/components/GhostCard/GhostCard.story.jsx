import React from 'react'
import { storiesOf } from '@storybook/react'

import GhostCard from '.'

storiesOf('GhostCard', module).add('Default', () => (
  <div className="p-16">
    <GhostCard width={500} height={200} />
  </div>
))
