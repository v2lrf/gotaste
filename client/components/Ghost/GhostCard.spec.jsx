import React from 'react'

import GhostCard from './GhostCard'

describe('GhostCard', () => {
  let component
  beforeEach(() => {
    component = shallow(<GhostCard height={200} />)
  })

  describe('Snapshots', () => {
    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
