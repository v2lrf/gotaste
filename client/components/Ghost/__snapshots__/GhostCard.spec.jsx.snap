// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GhostCard Snapshots should match the snapshot 1`] = `
<div
  aria-disabled={true}
  aria-label="Henter data"
  className="rounded shimmer"
  style={
    Object {
      "height": "200px",
      "width": "100%",
    }
  }
/>
`;
